### PR TITLE
Add missing decorator tests

### PR DESCRIPTION
**Pull request by pablothedude**
Some decorator tests needed to be implemented after the initial poc.
This was done to speedup initial development.

I skipped the Metadata tests as those implementations are going to be used as helper because they don't decorate any method so far.

https://www.pivotaltracker.com/story/show/169257771

### CODE REVIEW

File: src/OpenConext/EngineBlockBundle/Controller/MetadataController.php [hunk 1]
```diff
@@ -67,7 +67,6 @@ public function idpMetadataAction($keyId = null)
 
         $metadataXml = $this->metadataService->metadataForIdp(
             $this->getAbsoluteUrlForRoute('metadata_idp'),
```
[the remark this answers is not in the document]
Comment (pablothedude) [reply src/OpenConext/EngineBlockBundle/Controller/MetadataController.php]: Good suggestion!

File: src/OpenConext/EngineBlock/Metadata/Factory/Decorator/ServiceProviderProxy.php [hunk 2]
```diff
@@ -37,16 +40,22 @@ class ServiceProviderProxy extends AbstractServiceProvider
      * @var AttributesMetadata
      */
     private $attributes;
+    /**
+     * @var UrlProvider
+     */
+    private $urlProvider;
 
     public function __construct(
```
[the remark this answers is not in the document]
Comment (MKodde) [reply src/OpenConext/EngineBlock/Metadata/Factory/Decorator/ServiceProviderProxy.php]: I don't see the point to keep it in place. Once we start migrating other code from Corto that does require that visitor logic, we can always implement it then. For now, we don't use it, so there is no point in having to maintain this.

> Currently it is used to have all logic concerning the EB SP in one place in the domain layer

That logic should be on the EB SP decorator instance right?

[the remark this answers is not in the document]
Comment (pablothedude) [reply src/OpenConext/EngineBlock/Metadata/Factory/Decorator/ServiceProviderProxy.php]: What do you mean by 'EB SP decorator instance' or do you want me to rename the `ServiceProviderProxy` to `ServiceProviderEngineBlock`?

[the remark this answers is not in the document]
Comment (MKodde) [reply src/OpenConext/EngineBlock/Metadata/Factory/Decorator/ServiceProviderProxy.php]: No, move all the SPProxy logic to the EngineBlockServiceProviderMetadata decorator, as we already have a decorator that represents the EB SP entity.

[the remark this answers is not in the document]
Comment (pablothedude) [reply src/OpenConext/EngineBlock/Metadata/Factory/Decorator/ServiceProviderProxy.php]: Discussed IRL and I will create a separate PR to change the name of the SP-proxy to reflect it's usage as EB SP better and the Metadata to be a helper only for templating.

[the remark this answers is not in the document]
Comment (MKodde) [reply src/OpenConext/EngineBlock/Metadata/Factory/Decorator/ServiceProviderProxy.php]: Thanks!

**Pull request merged**